packages/fhir-router/src/repo.ts

```diff
@@ -457,10 +474,24 @@ export class MemoryRepository extends FhirRepository<undefined> {
     const { resourceType } = searchRequest;
     const resources = this.resources.get(resourceType) ?? new Map();
     const result = [];
-    for (const resource of resources.values()) {
-      if (matchesSearchRequest(resource, searchRequest)) {
-        result.push(resource);
+    try {
+      validateSearchResourceTypes(this, searchRequest);
```

Comment:
Moved resource type validation here to mirror the server respository implementation